core/src/SYCL/Kokkos_SYCL_Parallel_Scan.hpp

```diff
@@ -88,14 +88,7 @@ class ParallelScanSYCLBase {
     // FIXME_SYCL optimize
     constexpr size_t wgroup_size = 32;
     auto n_wgroups               = (size + wgroup_size - 1) / wgroup_size;
-
-    // FIXME_SYCL The allocation should be handled by the execution space
-    auto deleter = [&q](value_type* ptr) { sycl::free(ptr, q); };
-    std::unique_ptr<value_type[], decltype(deleter)> group_results_memory(
-        static_cast<pointer_type>(sycl::malloc(sizeof(value_type) * n_wgroups,
-                                               q, sycl::usm::alloc::shared)),
-        deleter);
-    auto group_results = group_results_memory.get();
+    auto group_results           = global_mem + size;
```

Comment:
I prefer `pointer_type` to `auto` here, as that emphasizes to the reader that pointer math is going on here.

Reply:
See a1c7fd008d86a785acbc70003d95cc843a42720c.